server/webhook.go

```diff
@@ -205,3 +213,192 @@ func (p *Plugin) getSubscriptionsWebhookURL(instanceID types.ID) string {
+		return
+	}
+
+	watcherUsers, err := client.GetWatchers(instanceID.String(), wh.Issue.ID, connection)
+	if err != nil {
+		p.errorf("error while getting watchers for issue , err : %v", err)
+		return
+	}
+
+	for _, watcherUser := range watcherUsers.Watchers {
+		if wh.checkNotificationAlreadyExist(watcherUser.DisplayName, watcherUser.AccountID) {
+			return
+		}
+
+		whUserNotification := webhookUserNotification{
+			jiraUsername:  watcherUser.DisplayName,
+			jiraAccountID: watcherUser.AccountID,
+			message:       commentMessage,
+			postType:      PostTypeComment,
+			commentSelf:   wh.JiraWebhook.Comment.Self,
+		}
+
+		wh.notifications = append(wh.notifications, whUserNotification)
+	}
+}
+
+func (wh *webhook) applyReporterNotification(p *Plugin, instanceID types.ID, reporter *jira.User) {
+	if !wh.eventTypes.ContainsAny("event_created_comment") {
+		return
+	}
+
+	jwhook := wh.JiraWebhook
+	if reporter == nil ||
+		(reporter.Name != "" && reporter.Name == jwhook.User.Name) ||
+		(reporter.AccountID != "" && reporter.AccountID == jwhook.Comment.UpdateAuthor.AccountID) {
+		return
+	}
+
+	if wh.checkNotificationAlreadyExist(reporter.Name, reporter.AccountID) {
+		return
+	}
+
+	commentAuthor := mdUser(&jwhook.Comment.UpdateAuthor)
+
+	commentMessage := fmt.Sprintf("%s **commented** on %s:\n>%s", commentAuthor, jwhook.mdKeySummaryLink(), jwhook.Comment.Body)
+
+	c, err := wh.GetUserSetting(p, instanceID, reporter.Name, reporter.AccountID)
+	if err != nil || c.Settings == nil || !c.Settings.ShouldReceiveReporterNotifications() {
+		return
+	}
+
+	wh.notifications = append(wh.notifications, webhookUserNotification{
+		jiraUsername:  reporter.Name,
+		jiraAccountID: reporter.AccountID,
+		message:       commentMessage,
+		postType:      PostTypeComment,
+		commentSelf:   jwhook.Comment.Self,
+	})
+}
+
+func (wh *webhook) checkNotificationAlreadyExist(username, accountID string) bool {
+	for _, val := range wh.notifications {
+		if val.jiraUsername == username && val.jiraAccountID == accountID {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (wh *webhook) GetUserSetting(p *Plugin, instanceID types.ID, jiraAccountID, jiraUsername string) (*Connection, error) {
+	var err error
+	instance, err := p.instanceStore.LoadInstance(instanceID)
+	if err != nil {
+		return nil, err
+	}
+	var mattermostUserID types.ID
+	if jiraAccountID != "" {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), jiraAccountID)
+	} else {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), jiraUsername)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := p.userStore.LoadConnection(instanceID, mattermostUserID)
+	if err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+func (s *ConnectionSettings) ShouldReceiveAssigneeNotifications() bool {
+	if s.SendNotificationsForAssignee != nil {
+		return *s.SendNotificationsForAssignee
+	}
+
+	// Check old setting for backwards compatibility
```

Comment:
Having this comment in each of these functions seems over-verbose to me. 0/5 to remove the comments